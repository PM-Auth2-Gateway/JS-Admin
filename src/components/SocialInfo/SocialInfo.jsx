import React from "react";
import classNames from "classnames";
import {Container, FormControl, FormGroup, FormLabel, Row, Col, FormText, Button} from "react-bootstrap";

import styles from "./SocialInfo.module.scss";
import socialIcon from "../../assets/google-avatar.png";

const SocialInfo = () => {
  return (
    <Container>
      <div className={styles.socialPreview}>
        <img src={socialIcon} className={styles.socialIcon} alt="icon"/>
        <h2>Social network name</h2>{/*TODO get from back*/}
      </div>
      <Row className={classNames("justify-content-center", styles.form)}>
        <Col lg={6} md={8}>
          {/*TODO get required field from back*/}
          <FormGroup>
            <FormLabel>Social Name</FormLabel>
            <FormControl type={"text"} readOnly defaultValue={"sadasdsadsad"}/>
          </FormGroup>
          <FormGroup>
            <FormLabel>Client ID</FormLabel>
            <FormControl type={"text"}/>
          </FormGroup>
          <FormGroup>
            <FormLabel>Client Secret</FormLabel>
            <FormControl type={"text"}/>
          </FormGroup>
          <FormGroup>
            <FormLabel>Allowed Mobile Client IDs</FormLabel>
            <FormControl as={'textarea'} rows={3}/>
            <FormText className={'text-muted'}>You can specify multiple valid client IDs by comma-separating
              them</FormText>
          </FormGroup>

          <Button variant={"info"}>Update</Button>
        </Col>
      </Row>
    </Container>
  )
}

export default SocialInfo;
