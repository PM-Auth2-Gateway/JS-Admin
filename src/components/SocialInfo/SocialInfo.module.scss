.socialPreview {
  display: flex;
  align-items: flex-start;
  gap: 24px;
  margin-bottom: 48px;

  .socialIcon {
    width: 64px;

  }
}

.form {
  border: 1px solid #c9cace;
  padding: 32px;
  border-radius: 6px;
}
