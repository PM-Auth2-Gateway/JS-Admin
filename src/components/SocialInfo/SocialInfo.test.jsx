import React from "react";
import {shallow} from 'enzyme';

import SocialInfo from "./SocialInfo";

describe("SocialInfo Component", () => {
  it('should render SocialInfo Component', function () {

    const component = shallow(<SocialInfo />);

    expect(component).toMatchSnapshot();
  });
})
