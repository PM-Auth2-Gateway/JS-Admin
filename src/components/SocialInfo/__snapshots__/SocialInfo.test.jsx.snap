// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialInfo Component should render SocialInfo Component 1`] = `
<Container
  fluid={false}
>
  <div
    className="socialPreview"
  >
    <img
      alt="icon"
      className="socialIcon"
      src="google-avatar.png"
    />
    <h2>
      Social network name
    </h2>
  </div>
  <Row
    className="form justify-content-center"
    noGutters={false}
  >
    <Col
      xs={5}
    >
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Social Name
        </FormLabel>
        <FormControl
          defaultValue="sadasdsadsad"
          readOnly={true}
          type="text"
        />
      </FormGroup>
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Client ID
        </FormLabel>
        <FormControl
          type="text"
        />
      </FormGroup>
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Client Secret
        </FormLabel>
        <FormControl
          type="text"
        />
      </FormGroup>
      <FormGroup>
        <FormLabel
          column={false}
          srOnly={false}
        >
          Allowed Mobile Client IDs
        </FormLabel>
        <FormControl
          as="textarea"
          rows={3}
        />
        <FormText
          className="text-muted"
        >
          You can specify multiple valid client IDs by comma-separating them
        </FormText>
      </FormGroup>
      <Button
        active={false}
        disabled={false}
        variant="info"
      >
        Update
      </Button>
    </Col>
  </Row>
</Container>
`;
